[Bug Report] SyncVectorEnv type-hint bug

### Describe the bug

In [`gymnasium/vector/sync_vector_env.py` Line32](https://github.com/Farama-Foundation/Gymnasium/blob/main/gymnasium/vector/sync_vector_env.py#L32), `env_fns` of type-hint is `Iterator[Callable[[], Env]]`.
The type-hint should be `Iterable`.

Ref [docs of mypy](https://mypy.readthedocs.io/en/stable/protocols.html?highlight=Iterator#iteration-protocols)

- `Iterator` needs to implement `__next__` and `__iter__` functions
- `Iterable` only needs to implement the `__iter__` function

`env_fns` is only used in [`gymnasium/vector/sync_vector_env.py` Line 52](https://github.com/Farama-Foundation/Gymnasium/blob/main/gymnasium/vector/sync_vector_env.py#L52) (`self.envs = [env_fn() for env_fn in env_fns]`). For this line of code, the `Iterable` type is sufficient.

`Iterator` is more restrictive than `Iterable`, causing `mypy` to throw an error when you pass it to `list`.


### Code example

```shell
import gymnasium as gym

envs = gym.vector.SyncVectorEnv([lambda: gym.make("CartPole-v1") for _ in range(3)])
```
mypy error report:

```sh
❯ mypy a.py                           
a.py:3: error: Argument 1 to "SyncVectorEnv" has incompatible type "List[Callable[[], Env[Any, Any]]]"; expected "Iterator[Callable[[], Env[Any, Any]]]"  [arg-type]
a.py:3: note: "list" is missing following "Iterator" protocol member:
a.py:3: note:     __next__
Found 1 error in 1 file (checked 1 source file)
```


### System info

- pip install
- gymnasium0.26.3
- python3.9.13
- mypy0.991
- macOS13.0.1

### Additional context

_No response_

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/Gymnasium/issues) in the repo

--- a/gymnasium/vector/sync_vector_env.py
+++ b/gymnasium/vector/sync_vector_env.py
@@ -1,6 +1,6 @@
 """A synchronous vector environment."""
 from copy import deepcopy
-from typing import Any, Callable, Iterator, List, Optional, Sequence, Union
+from typing import Any, Callable, Iterable, List, Optional, Sequence, Union
 
 import numpy as np
 
@@ -29,7 +29,7 @@ class SyncVectorEnv(VectorEnv):
 
     def __init__(
         self,
-        env_fns: Iterator[Callable[[], Env]],
+        env_fns: Iterable[Callable[[], Env]],
         observation_space: Space = None,
         action_space: Space = None,
         copy: bool = True,

